src/internal_modules/roc_fec/reader.cpp

```diff
@@ -789,5 +804,24 @@ void Reader::drop_repair_packets_from_prev_blocks_() {
     }
 }
 
+void Reader::update_block_duration_(const packet::PacketPtr& ptr) {
+    if (!ptr->rtp()) {
+        return;
+    }
+    packet::stream_timestamp_diff_t block_dur = 0;
+    if (prev_block_timestamp_valid_) {
+        block_dur = packet::stream_timestamp_diff(ptr->rtp()->stream_timestamp,
+                                                  prev_block_timestamp_);
+    }
+    roc_panic_if_msg(block_dur < 0, "fec reader: negative block duration");
```

Comment:
I think panic is not appropriate here, because corrupted/malformed packets would lead to a crash?

Instead we can skip the check. Probably also log to LogTrace.